mne/datasets/utils.py

```diff
@@ -384,7 +384,7 @@ def _data_path(path=None, force_update=False, update_path=True, download=True,
     want_version = _FAKE_VERSION if name == 'fake' else want_version
     if not need_download and want_version is not None:
         data_version = _dataset_version(folder_path[0], name)
-        need_download = data_version != want_version
+        need_download = LooseVersion(data_version) < LooseVersion(want_version)
```

Comment:
Added this because in general it should be safe enough to use a newer version with old code (it's rare that it's not). Otherwise we're stuck in a position where for example I couldn't test your PR because I had a newer version from a PR that has been merged to master, which is not great.